### PR TITLE
Add `Debug` impls for public structs in Parquet crate

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

N/A

# Rationale for this change
 
This caught me when working on https://github.com/influxdata/influxdb_iox/pull/7855

I had to manually derive a `Debug` impl for a structure because the `ArrowWriter` didn't

# What changes are included in this PR?

1. Add `Debug` impls for public structs in Parquet crate

Note that PR quickly got out of hand, but since I was in here I figured I would simply bash it out

# Are there any user-facing changes?
More `Debug`!

### CODE REVIEW

File: parquet/src/arrow/array_reader/null_array.rs [hunk 1]
```diff
@@ -42,6 +43,17 @@ where
     record_reader: RecordReader<T>,
 }
 
+impl<T> Debug for NullArrayReader<T>
+where
+    T: DataType,
+    T::T: ScalarValue,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("NullArrayReader<T>")
```
Comment (alamb) [inline parquet/src/arrow/array_reader/null_array.rs:52]: I freely admit I got lazy for some of these impls...

File: parquet/src/record/api.rs [hunk 2]
```diff
@@ -191,7 +192,7 @@ macro_rules! row_complex_accessor {
 
 impl RowFormatter for Row {
     /// Get Display reference for a given field.
-    fn fmt(&self, i: usize) -> &dyn fmt::Display {
+    fn fmt_for_display(&self, i: usize) -> &dyn fmt::Display {
```
Comment (alamb) [inline parquet/src/record/api.rs:195]: This is a crate private trait and was shadowing the `fmt` in `Debug` so I renamed it

**Comment by tustvold** [comment]
Could we just manually implement Debug for the public components, this seems like an inordinate amount of complexity that isn't really useful to print anyway?

**Comment by alamb** [comment]
Will make a new PR with just the debug impls I want

**Comment by alamb** [comment]
https://github.com/apache/arrow-rs/pull/4278

**Pull request closed without merging**